### PR TITLE
[FPGA] Add example script for VGG Convolutions

### PR DESCRIPTION
**Pull request by manuelburger**
Adds example script for running the convolutional part of the VGG16 architecture on FPGA in the `fpga` test directory.

### CODE REVIEW

File: tests/pytorch/fpga/vgg_conv_optimization_example.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        "tile": 196,
+        "vec": 2
+    }]
+}
+
+
+# ----------------------------------------
+# Memory bank RR-interleaving
+# ----------------------------------------
+def fpga_rr_interleave_containers_to_banks(sdfg: SDFG, num_banks: int = 4):
```
Comment (TizianoDeMatteis) [inline tests/pytorch/fpga/vgg_conv_optimization_example.py:94]: Why are you not using the methods offered by dace?

File: tests/pytorch/fpga/vgg_conv_optimization_example.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+# Random numbers
+import random
+
+torch.manual_seed(0)
+random.seed(0)
+np.random.seed(0)
+
+# ----------------------------------------
+# Node Mappings
```
Comment (TizianoDeMatteis) [inline tests/pytorch/fpga/vgg_conv_optimization_example.py:45]: Could you please explain what are these mappings?

File: tests/pytorch/fpga/vgg_conv_optimization_example.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+                desc, dt.Stream
+        ) and desc.storage == dtypes.StorageType.FPGA_Global and desc.transient:
+            desc.location["bank"] = i % num_banks
+            num_allocated[i % num_banks] = num_allocated[i % num_banks] + 1
+            i = i + 1
+
+    return num_allocated
+
+
+# def expand_library_nodes_kwargs(sdfg, recursive=True, **kwargs):
```
Comment (TizianoDeMatteis) [inline tests/pytorch/fpga/vgg_conv_optimization_example.py:119]: Remove it

File: tests/pytorch/fpga/vgg_conv_optimization_example.py (hunk 1 above)
Comment (manuelburger) [reply tests/pytorch/fpga/vgg_conv_optimization_example.py:94]: Using the DaCe implementation now

File: tests/pytorch/fpga/vgg_conv_optimization_example.py (hunk 2 above)
Comment (manuelburger) [reply tests/pytorch/fpga/vgg_conv_optimization_example.py:45]: Added an explanation

File: tests/pytorch/fpga/vgg_conv_optimization_example.py (hunk 3 above)
Comment (manuelburger) [reply tests/pytorch/fpga/vgg_conv_optimization_example.py:119]: removed

**Comment by manuelburger** [comment]
> You could also consider adding it as an example script to the documentation: https://daceml.readthedocs.io/en/latest/auto_examples/index.html
> 
> I think the docs build is currently broken, but I'll fix that soon.

I could yes. The current location was just where a similar example script for LeNet is and also all other FPGA related material, so that was the reason for me putting it there. @TizianoDeMatteis any preferences?

**Comment by TizianoDeMatteis** [comment]
> > You could also consider adding it as an example script to the documentation: https://daceml.readthedocs.io/en/latest/auto_examples/index.html
> > I think the docs build is currently broken, but I'll fix that soon.
> 
> I could yes. The current location was just where a similar example script for LeNet is and also all other FPGA related material, so that was the reason for me putting it there. @TizianoDeMatteis any preferences?

@orausch Since this is not the full VGG (normalization is missing) I will keep it there and move it to the example later. Is this ok for you?

**Comment by orausch** [comment]
Sure

**Review by TizianoDeMatteis — APPROVED** [review]

**Pull request merged**